### PR TITLE
redist 3.1

### PR DESCRIPTION
**Pull request by CoryMcCartan**
Ongoing draft PR for redist 3.1.  See milestone 3.1 for details. So far

- [x] remove deprecated arguments and functions
- [x] improve SMC/MS county performance
- [x] interactive plotting
- [x] misc. plotting and interface improvements
- [x] parallel redist_mergesplit

### CODE REVIEW

**Comment by christopherkenny** [comment]
@CoryMcCartan, good to merge?

**Comment by CoryMcCartan** [comment]
We ran into that missing `geometry` column in plotting that we should fix.

**Comment by christopherkenny** [comment]
Okay, I'll look at replacing calls to "geometry" with the sf_col attribute.

File: src/smc.cpp [hunk 1]
```diff
@@ -186,10 +213,11 @@ void split_maps(const Graph &g, const uvec &counties, Multigraph &cg,
         }
         // `lower_s` now contains the population of the newly-split district
         pop_left_new(i) = pop_left(idx) - lower_s;
-        double pop_pen = sqrt((double) n_distr - 2) * log(std::fabs(lower_s - target)/target);
-        log_temper_new(i) = log_temper(idx) - pop_temper*pop_pen;
+        double dev = std::fabs(lower_s - target)/target;
+        double pop_pen = sqrt((double) n_distr - 2) * log(1e-12 + dev);
```
[the remark this answers is not in the document]
Comment (CoryMcCartan) [reply src/smc.cpp:217]: No. It's not a long double

**Comment by christopherkenny** [comment]
The issue is appearing across all OSes from time to time, just consistently on Linux noLD and Solaris. Looking through the old GitHub actions, it’s failed on every system at some point in time over the last 3 months due to a mysterious timeout.

**Pull request merged**